lib/client.js

```diff
@@ -1361,16 +1361,11 @@ function write (client, request) {
     return false
   }
```

Comment:
Should be enough to just do:

```
if (reset) {
  socket[kReset] = true
}
```

I don't think you need the rest of the changes here.

Reply:
Sadly it doesn't work without it, as we need to use the `Request#reset` to see if this request in specific will be reset after response. If relying only on the `pipelining` option, we'll ignore the Request options and continue sending the `connection: keep-alive` header.

Or do you refer to something else? 🤔

Reply:
https://github.com/nodejs/undici/pull/1846

Reply:
Got it, let me rebase and adjust 👍